core/src/main/java/brooklyn/config/external/ExternalConfigSupplier.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package brooklyn.config.external;
+
+/**
+ * Provider of "externalised" entity configuration that is resolved at runtime.
+ */
+public interface ExternalConfigSupplier {
+
+    String getName();
+    String get(String key);
```

Comment:
As stated in your doc:

    The supplier’s get()-semantics should be clearly described in the javadoc of the implementation class.

Can you add javadoc to that effect please.